app/controllers/caterings_controller.rb

```diff
@@ -0,0 +1,46 @@
+class CateringsController < ApplicationController
+
+	def index
+		@caterings = Catering.all
+	end
+
+	def show
+		begin @catering = Catering.find(params[:id])
+		rescue ActiveRecord::RecordNotFound
+			flash.now.alert = "This Catering does not exist"
```

Comment:
Cant we use our private 'record_not_found' function? This code looks like doing the same thing as this function.
Also you are setting the default action for rescuing the RecordNotFound exception on line #40  and then rescue it explicitly with this code (#10 - #11).